setup.py

```diff
@@ -16,6 +16,22 @@
 except ImportError:
     tagger = egg_info_cmd.egg_info
 
+install_requires=[
+    'requests',
+    'PyYAML',
+    'rdflib >= 4.1.0',
+    'rdflib-jsonld >= 0.3.0',
+    'mistune'
+]
+
+avro_package = 'avro'
+if sys.version_info >= (3,):
+    avro_package += '-python3'
+install_requires.append(avro_package)
+
+if sys.version_info >= (2, 7):
+    install_requires.append('typing')
+
```

Comment:
So this style of declaring dependencies works for ``python setup.py`` but when doing ``pip install`` from PyPI I've noticed some subtle problems in the past (maybe it was user error though) - the wheel would have the dependencies for the version of Python it was created with - not the version of Python it is being installed into.

Here is how I "fixed' this in planemo with ``extras_requires``.

https://github.com/galaxyproject/planemo/commit/006917ff01635208e4bfda66ccfadfe45dee8a77